src/controllers/stallController.ts

```diff
@@ -139,16 +135,55 @@ async function destroyStall(req: Request, res: Response, next: NextFunction) {
   }
 }
 
+/**
+ * Retrieves all stalls that match a given set of ids.
+ * @param ids Ids of stalls to be retrieved.
+ */
 async function findStallsByIds(ids: number[]) {
   const stalls = await Stall.findAll({
     include: getStallInclude(),
     where: {
       id: ids,
     },
   });
+
+  // To obtain all stall ratings
+  const ratings = await Review.findAll({
+    where: {
+      stallId: ids,
+    },
+    attributes: [
+      'stallId',
+      [Sequelize.fn('ROUND', Sequelize.fn('AVG', Sequelize.col('rating')), 2), 'rating'],
+    ],
+    group: ['stallId'],
+  });
+
+  stalls.map(async (stall: Stall) => {
+    const filteredRating = ratings.filter(rating => rating.stallId === stall.id);
+    const rating: number = filteredRating.length ? filteredRating[0].rating : 0;
+    await stall.setDataValue('rating', rating);
+  });
+
   return stalls;
 }
 
+/**
+ * Formats stall information to display them on cards.
+ * @param stalls Sequelize instances of stalls to be formmatted.
+ */
+function mapStallToCard(stalls: Stall[]) {
+  const updatedStalls = stalls.map(stall => {
+    const jsonStall = JSON.parse(JSON.stringify(stall));
+
+    jsonStall['Categories'] = jsonStall['Categories'].map((category: Category) => category['name']);
+    const propertiesToDelete = ['description', 'contactNo', 'unitNo', 'Products', 'Reviews'];
```

Comment:
You can set the properties you need in `include` under the key `attribute`.

Reply:
If you are doing this, its better to standardise throughout all the responses of `Stall`. We could probably do this in another PR.

Reply:
Got it, ill add it to the agenda for next week's sprint